### DISCUSSION TITLE
Parsing JSON Files for tailwind classes

### DISCUSSION THREAD

**Discussion by ng-druid** [tailwindlabs/tailwindcss#9078] [General]
We dynamically generate html pages from json files.

An example of these json files are here.

https://github.com/rollthecloudinc/ipe-objects-prod/tree/master/panelpage

Is it possible to point tail wind at this directory and have it extract and build a style sheet or does it only work for html files.

We do pre-render these json files to html files. However, the pre-render process doesn't always work as expected leaving out some physical html that is part of the page. To fill this void for now it would be best to analyze the json files themselves.

This is our build process.

https://github.com/rollthecloudinc/druid/blob/master/.github/workflows/druid-workflow.yml

Ideally we would just be able to tell tail wind to look inside both dist (html) and projects/spear/src/assets/objects/panelpage (json) directories.

Thanks

**Comment by wongjn** [comment]
Yep should work! Tailwind's content scanning is file-type agnostic – it just scans for text content and matches classes that it knows about as text strings. So, as long as Tailwind classes you use exist as pure strings inside the JSON files, your Tailwind classes should be generated fine.
